src/coreclr/System.Private.CoreLib/src/System/RuntimeType.CoreCLR.cs

```diff
@@ -4003,6 +4008,11 @@ internal object GetUninitializedObject()
                 throw new MissingMethodException(SR.Format(SR.Arg_NoDefCTor, this));
             }
 
+            if (IsByRefLike)
```

Comment:
This should not be needed. This method should not be ever called on value types. (We can add asserts.)